open_event/static/js/admin/event/scheduling.js

```diff
@@ -0,0 +1,502 @@
+                $sessionElement.css({
+                    "-webkit-transform": "",
+                    "transform": "",
+                    "background-color": ""
+                }).removeData("x").removeData("y").tooltip("hide").attr("data-original-title", "");
+            }
+        }
+    });
+}
+
+function minutesToPixels(minutes, forTop) {
+    minutes = Math.abs(minutes);
+    if (forTop) {
+        return ((minutes / time.unit.minutes) * time.unit.pixels) + time.unit.pixels;
+    } else {
+        return (minutes / time.unit.minutes) * time.unit.pixels;
+    }
+}
+
+function pixelsToMinutes(pixels, fromTop) {
+    pixels = Math.abs(pixels);
+    if (fromTop) {
+        return ((pixels - time.unit.pixels) / time.unit.pixels) * time.unit.minutes;
+    } else {
+        return (pixels / time.unit.pixels) * time.unit.minutes;
+    }
+}
+
+var days = [];
+var sessionsStore = [];
+var tracksStore = [];
+var currentUnscheduledStore = [];
+
+function processTrackSession(tracks, sessions, callback) {
+
+    var topTime = moment.utc({hour: time.start.hours, minute: time.start.minutes});
+
+    _.each(sessions, function (session) {
+
+        var startTime = moment.utc(session.start_time);
+        var endTime = moment.utc(session.end_time);
+        var duration = moment.duration(endTime.diff(startTime));
+
+        var top = minutesToPixels(moment.duration(moment.utc({
+            hour: startTime.hours(),
+            minute: startTime.minutes()
+        }).diff(topTime)).asMinutes(), true);
+
+        var dayString = startTime.format("Do MMMM YYYY"); // formatted as eg. 2nd May
+
+        if (!_.includes(days, dayString)) {
+            days.push(dayString);
+        }
+
+        var sessionObject = {
+            id: session.id,
+            title: session.title,
+            top: top,
+            duration: Math.abs(duration.asMinutes()),
+            track_id: session.track.id,
+            start_time: startTime.format("HH:mm"),
+            end_time: endTime.format("HH:mm")
+        };
+
+        var dayIndex = _.indexOf(days, dayString);
+
+        if (_.isArray(sessionsStore[dayIndex])) {
+            sessionsStore[dayIndex].push(sessionObject);
+        } else {
+            sessionsStore[dayIndex] = [sessionObject]
+        }
+
+    });
+
+    _.each(tracks, function (track) {
+        var tracksObject = {
+            name: track.name,
+            id: track.id,
+            count: track.sessions.length
+        };
+        if (!_.includes(tracksStore, tracksObject)) {
+            tracksStore.push(tracksObject);
+        }
+    });
+
+
+    loadDateButtons();
+    callback();
+}
+
+function loadDateButtons() {
+    var dayButtonTemplate = $("#date-change-button-template").html();
+    var $dayButtonsHolder = $(".date-change-btn-holder");
+    var sortedDays = days.sort();
+    _.each(sortedDays, function (day, index) {
+        var $dayButton = $(dayButtonTemplate);
+        if (index == 0) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2651583495/issues/source?bid=3233864&fileBranchId=3324174#l336)